### PR TITLE
pkp/pkp-lib#1244 Strip tags from record title and fields

### PR DESCRIPTION
**Pull request by mfelczak**

### CODE REVIEW

**Comment by asmecher** [comment]
Do you have an example where this is needed? Generally `escape` is the best approach, I think -- consider a title with `<` or `>` legitimately appearing in it. Running it through `strip_tags` first would remove that.

**Comment by mfelczak** [comment]
It comes up occasionally in harvested metadata (screenshots attached).

I think we'd need both `<` and `>` to appear in a title or field to get filtered out (vs. just one or the other).

I took a look at how we handle DC metadata in OJS and we're currently stripping and escaping, i.e.
`<meta name="DC.Title" content="{$article->getLocalizedTitle()|strip_tags|escape}"/>`

**Comment by asmecher** [comment]
That `if gte mso 9` glob of pure evil is a cut/paste job from Microsoft Word and it's horribly nonstandard. I don't know if we really want to try to clean it up.

The `<meta name="DC.Title" content="...">` isn't what's feeding the OAI interface -- it's the equivalent of a pure `escape` modifier (though it's not coded in Smarty).

I still think it's more important to support titles with < and > (and &) predictably than to support HTML entities in those fields. "This & That" and "Statistical Significance < 0.5" are much more plausible/standard titles than "Something <strong>Containg</strong> HTML". And nothing we've done in the OJS interface encourages HTML in title fields.

I'm speaking purely of titles here -- abstracts do/should contain HTML, so obviously that's different.

**Comment by mfelczak** [comment]
That makes sense to me.

This particular case may actually be limited to the MARC harvest and display. The examples use the 'summary' field as the title when the title is missing. The 'summary' fields include a mix of evil globs and valid HTML.

How about the following then:
- don't use the 'summary' field as the title, display a dash if the title is missing;
- escape the title;
- strip_unsafe_html all other fields to clean-up the current display (screenshot attached)

**Comment by asmecher** [comment]
Yes, that sounds logical to me. Thanks, @mfelczak.

**Comment by mfelczak** [comment]
Thanks @asmecher -- I've updated the PR.

File: plugins/schemas/marc/record.tpl [hunk 1]
```diff
@@ -33,7 +33,7 @@
 		<tr valign="top">
 			<td>{if $name}{translate key="plugins.schemas.marc.fields.$name.name"}{else}&mdash;{/if}</td>
 			<td>
-				{foreach from=$entry item=i1val key=i1}{foreach from=$i1val item=i2val key=i2}{foreach from=$i2val item=codeval key=code}{foreach from=$codeval item=value}{$value|escape|default:"&mdash;"}<br/>{/foreach}{/foreach}{/foreach}{/foreach}
+				{foreach from=$entry item=i1val key=i1}{foreach from=$i1val item=i2val key=i2}{foreach from=$i2val item=codeval key=code}{foreach from=$codeval item=value}{$value|trim|nl2br|strip_unsafe_html|default:"&mdash;"}<br/>{/foreach}{/foreach}{/foreach}{/foreach}
```
Comment (asmecher) [inline plugins/schemas/marc/record.tpl]: I'm not sure about this one -- previously we treated all fields as plaintext, and this change makes them all HTML-capable. I suspect selective treatment is the only way to get this right.

File: plugins/schemas/mods/record.tpl [hunk 2]
```diff
@@ -93,7 +93,7 @@
 		{if $contents.$nodeName}
 			<tr valign="top">
 				<td class="label">{if $nodeName}{translate key="plugins.schemas.mods.fields.$nodeName.name"}{else}&mdash;{/if}</td>
-				<td class="value">{$contents.$nodeName|escape|nl2br}</td>
+				<td class="value">{$contents.$nodeName|trim|nl2br|strip_unsafe_html}</td>
```
Comment (asmecher) [inline plugins/schemas/mods/record.tpl]: (as above)

**Comment by mfelczak** [comment]
Thanks again, @asmecher.

PR updated with selective treatment for abstracts, all other fields escaped as before.

**Comment by asmecher** [comment]
Thanks, @mfelczak, looks good.

**Pull request merged**